scripts/gha/test_simulator.py

```diff
@@ -254,6 +311,135 @@ def _read_file(path):
   return test_result
 
 
+def _check_java_version():
+  command = "java -version 2>&1 | awk -F[\\\"_] 'NR==1{print $2}'"
+  logging.info("Get java version: %s", command)
+  result = subprocess.Popen(command, universal_newlines=True, shell=True, stdout=subprocess.PIPE)
```

Comment:
nit: Recommended way to do this is with `subprocess.run`.

Reply:
`subprocess.run` doesn’t work with pipes. Please correct me if I was wrong